src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts

```diff
@@ -51,20 +58,44 @@ export class AutofillProjectFormComponent implements OnInit {
 
     if (this.publicationDoiCtrl.value) {
       const doi = this.publicationDoiCtrl.value;
-      this.doesProjectWithDoiExist(doi).pipe(
-        filter(projectExists => projectExists === false),
-        switchMapTo(this.doesDoiExist(doi)),
-        filter(doiExists => doiExists)
-      ).subscribe(() => {
-          this.createProject(doi);
+      this.loading = true;
+      this.getProjectsWithDOI(doi).pipe(
+        switchMap(projects =>
+          this.doesDoiExist(doi).pipe(
+            map(doiExists => ({ doiExists, projects }))
+          )
```

Comment:
suggestion, the pipe should be:
```
switchMap(projects =>this.doesDoiExist(doi)),
map(doiExists => ({ doiExists, projects }))
```

Reply:
This won't work. The switchmap is creating a new observable from `doesDoiExist` (inner) and mapping the result of that to the outer observable (`getProjectsWithDOI`)

Reply:
okay in that case I think the following is better since in your current implementation you have to run both searches whether on not a project contains the DOI:
```
this.loading = true;
forkJoin(
    projects: this.getProjectsWithDOI(doi),
    doiExists: this.doesDoiExist(doi)
).subscribe(({projects, doiExists}) => {
```

Reply:
Yep this is much better, thanks!